src/filters.ts

```diff
@@ -188,10 +189,14 @@ export class MatchAttributes implements ItemFilter {
     const matches: Set<string> = new Set();
 
     for (const [name, value] of Object.entries(this.attributes)) {
+      const txLikeItem = item as MatchableTxLike;
```

Comment:
For the attributes filter, let's create a similar `MatchObjectAttributes` filter that can be applied to any object, but doesn't have the TX like specific logic. We can use it instead of `MatchAttributes` when instantiating matches in `createObjectFilter`.

Reply:
good idea

Reply:
I had that idea too, I would also want to use R.equals in that new MatchObjectAttributes for value comparisons if one wants to compare say array attribute values

Reply:
implemented